### ISSUE TITLE
TextInput disabled={true} triggers error Warning: Cannot update a component (`FormTabHeader`) while rendering a different component (`TextInput`)

### ISSUE THREAD

**Issue by panfiva** [marmelab/react-admin#9518]
After upgrade to "react-hook-form": "7.49.0", `<TextInput source="id" disabled={true} />` produces an error.

In included example, this error is not significant; however, in more dynamic forms with watches and validations, it triggers infinite refresh and browser freezing.

**What you were expecting:**
No error

**Steps to reproduce:**
- https://stackblitz.com/edit/github-sz4fgz?file=src%2Ftags%2FTagEdit.tsx,src%2Fposts%2FPostEdit.tsx
- Navigate to Posts and select a record to edit
- Observe error in browser console

```
Warning: Cannot update a component (`FormTabHeader`) while rendering a different component (`TextInput`). To locate the bad setState() call inside `TextInput`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render
    at TextInput (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:119624:25)
    at div
    at https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-SDUT2LJQ.js?v=92684455:2451:46
    at Box3 (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-SDUT2LJQ.js?v=92684455:4895:19)
    at SanitizedBox (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/src/posts/PostEdit.tsx?t=1702410744578:125:3)
    at div
    at https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-SDUT2LJQ.js?v=92684455:2451:46
    at Grid2 (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-SDUT2LJQ.js?v=92684455:6773:24)
    at FormGroupContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:16296:23)
    at FormTab (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:121572:24)
    at div
    at https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-SDUT2LJQ.js?v=92684455:2451:46
    at CardContent2 (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-RJZB3EAH.js?v=92684455:19796:17)
    at DefaultComponent2 (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:121434:23)
    at div
    at https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-SDUT2LJQ.js?v=92684455:2451:46
    at TabbedFormView (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:121378:24)
    at form
    at FormGroupsProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:15707:23)
    at FormProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-EOHCH6WT.js?v=92684455:80:11)
    at LabelPrefixContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:15009:21)
    at RecordContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:9535:23)
    at OptionalRecordContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:9929:20)
    at Form (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:16093:24)
    at TabbedForm (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:121642:26)
    at div
    at https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-SDUT2LJQ.js?v=92684455:2451:46
    at Paper2 (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-RJZB3EAH.js?v=92684455:1696:17)
    at https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-SDUT2LJQ.js?v=92684455:2451:46
    at Card2 (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-RJZB3EAH.js?v=92684455:19540:17)
    at div
    at div
    at https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-SDUT2LJQ.js?v=92684455:2451:46
    at EditView (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:121084:23)
    at RecordContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:9535:23)
    at SaveContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:9153:23)
    at EditContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:9962:23)
    at EditBase (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:9988:23)
    at Edit (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:121174:3)
    at PostEdit (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/src/posts/PostEdit.tsx?t=1702410744578:146:7)
    at RenderedRoute (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-UPAPYNEA.js?v=92684455:3421:5)
    at Routes (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-UPAPYNEA.js?v=92684455:3863:5)
    at ResourceContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:14471:23)
    at Resource (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:14477:23)
    at RenderedRoute (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-UPAPYNEA.js?v=92684455:3421:5)
    at Routes (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-UPAPYNEA.js?v=92684455:3863:5)
    at Suspense
    at ErrorBoundary2 (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:23902:35)
    at div
    at main
    at div
    at div
    at https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-SDUT2LJQ.js?v=92684455:2451:46
    at Layout (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:24298:20)
    at default
    at div
    at RenderedRoute (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-UPAPYNEA.js?v=92684455:3421:5)
    at Routes (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-UPAPYNEA.js?v=92684455:3863:5)
    at CoreAdminRoutes (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:14354:3)
    at RenderedRoute (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-UPAPYNEA.js?v=92684455:3421:5)
    at Routes (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-UPAPYNEA.js?v=92684455:3863:5)
    at CoreAdminUI (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:14418:20)
    at div
    at https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-SDUT2LJQ.js?v=92684455:2451:46
    at ScopedCssBaseline2 (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-RJZB3EAH.js?v=92684455:30294:17)
    at AdminUI (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:126168:18)
    at ThemeProvider2 (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-SDUT2LJQ.js?v=92684455:5299:5)
    at ThemeProvider3 (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-SDUT2LJQ.js?v=92684455:5353:5)
    at ThemeProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-MLFG6QWZ.js?v=92684455:278:12)
    at ThemeProvider2 (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:21651:23)
    at ResourceDefinitionContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:13987:18)
    at NotificationContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:5129:23)
    at I18nContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:15226:18)
    at Router (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-UPAPYNEA.js?v=92684455:3810:15)
    at HistoryRouter (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:5601:23)
    at InternalRouter (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:5641:23)
    at BasenameContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:5621:23)
    at AdminRouter (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:5627:22)
    at QueryClientProvider2 (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-CNPG7RK6.js?v=92684455:2852:25)
    at PreferencesEditorContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:13903:23)
    at StoreContextProvider (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:5035:20)
    at CoreAdminContext (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:14013:28)
    at AdminContext (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/chunk-IHU7MYMU.js?v=92684455:126204:21)
    at Admin (https://githubsz4fgz-guua--8080--d3416dfd.local-credentialless.webcontainer.io/node_modules/.vite/deps/react-admin.js?v=671d3fe4:664:28)
```

```
<Edit title={<PostTitle />} actions={<EditActions />}>
      <TabbedForm defaultValues={{ average_note: 0 }} warnWhenUnsavedChanges>
        <TabbedForm.Tab label="post.form.summary">
          <SanitizedBox
            display="flex"
            flexDirection="column"
            width="100%"
            justifyContent="space-between"
            fullWidth
          >
            <TextInput source="id" disabled={true} />
            <TextInput source="title" validate={required()} resettable />
          </SanitizedBox>
......
```

**Other information:**

**Environment**

* React-admin version:
* Last version that did not exhibit the issue (if applicable):
* React version:
* Browser:
* Stack trace (in case of a JS error):

**Comment by ns-vpanfilov** [comment]
It appears that the issue has been addressed in "react-hook-form": "7.49.1"

**Comment by panfiva** [comment]
It appears that the issue has been addressed in "react-hook-form": "7.49.1"
